current/advanced/codec.md

```diff
@@ -95,6 +95,13 @@ As an exception, in the case that the type is a boolean, then it is always one b
 - `0x01` if it is the `false` value.
 - `0x02` if it is the `true` value.
 
+### Results
+
+Results indicate whether certain operations were successful or unsuccessful. Each variant can contain additional data, such as a resulting value or an error message.
```

Comment:
How is the (optional?) error message encoded/provided? A few examples here would be useful I think.

Reply:
It depends on the type how they are encoded. Nothing special and nothing that needs to be documented here.

Reply:
@dvdplm At the beginning of this document it straight out says:

>  It is not self-describing in any way and assumes the decoding context has all type knowledge about the encoded data.

It's just a SCALE encoded type the decoder needs to know about.

Reply:
My point was not to debate how SCALE works. As it stands a developer that is unfamiliar with SCALE will not gain much info from this paragraph (I know because I am one such developer) and providing an example or two of how it works would be useful imo. If you are confident that the intended audience will understand and find it useful then I have no objections.

Reply:
We could maybe reuse the enum example and adjust it to:

```rust
enum Result {
  Ok(u8),
  Err(bool),
}
```

- `Ok(42)`: `0x002a`
- `Err(false)`: `0x0100`

If you think it should be there then feel free to add it.

Reply:
although, it can't hurt. I'll update this PR.

Reply:
@dvdplm Updated!